src/commands/router.ts

```diff
@@ -1,76 +1,106 @@
-import { Message, MessageEmbed } from 'discord.js';
+import { Interaction, Message, MessageEmbed, GuildMember } from 'discord.js';
 import config from '../config';
-import { getMissingPermissions, isAdmin } from '../utils';
+import { isAdmin } from '../utils';
 import CommandError from './CommandError';
 import commands from './instances';
-import * as parser from './parser';
+import { COUNTRIES } from '../utils/countries';
+import { ALL_METRICS } from '../utils';
 import { customCommands } from './CustomCommands';
-import { CustomCommand } from '../types';
 
-export function onError(message: Message, title: string, tip?: string): void {
-  const response = new MessageEmbed().setColor(config.visuals.red).setDescription(title);
-  message.channel.send(tip ? response.setFooter(tip) : response);
-}
-
-export async function onMessageReceived(message: Message): Promise<void> {
-  if (!parser.isValid(message)) {
+export function onError(options: {
+  message?: Message;
+  interaction?: Interaction;
+  title: string;
+  tip?: string;
+}): void {
+  const response = new MessageEmbed().setColor(config.visuals.red).setDescription(options.title);
+  response.setFooter({ text: options.tip ? options.tip : '' });
+  if (options.message) {
+    options.message.channel.send({ embeds: [response] });
     return;
+  } else if (options.interaction && options.interaction.isCommand()) {
+    options.interaction.reply({ embeds: [response] });
   }
+}
 
-  const parsed = await parser.parse(message);
-  const missingPermissions = getMissingPermissions(message.guild?.me);
+// Slash commands
+export async function onInteractionReceived(interaction: Interaction): Promise<void> {
+  if (interaction.isAutocomplete()) {
+    const focused = interaction.options.getFocused(true);
+    const currentValue = focused.value.toString();
+    if (focused.name === 'country') {
+      const options = COUNTRIES.filter(country =>
+        !currentValue
+          ? false
+          : [country.name.toLowerCase(), country.code.toLowerCase()].some(str =>
+              str.includes(currentValue.toLowerCase())
+            )
+      ).map(c => ({ name: c.name, value: c.code }));
```

Comment:
For a smarter autcomplete, I'd recommend `levenshtein` (fastest), or `fuse.js` (smartest)
Both of these will handle minor typos and return relevant results.
![image](https://user-images.githubusercontent.com/9059487/164754664-8f67a437-1cc3-410c-bfc1-4e81e08885b4.png)